src/lib/logical_query_plan/abstract_lqp_node.cpp

```diff
@@ -274,60 +274,80 @@ bool AbstractLQPNode::has_matching_unique_constraint(const ExpressionUnorderedSe
 }
 
 std::vector<FunctionalDependency> AbstractLQPNode::functional_dependencies() const {
-  // Gather input FDs from all input nodes
-  auto fds_left = std::vector<FunctionalDependency>();
-  auto fds_right = std::vector<FunctionalDependency>();
-  if (left_input()) {
-    fds_left = left_input()->functional_dependencies();
-  }
-  if (right_input()) {
-    fds_right = right_input()->functional_dependencies();
+  // (1) Gather FDs from previous nodes
```

Comment:
Explain that these exclude FDs based on unique constraints?

Reply:
_(1) Gather FDs from previous nodes that cannot be derived from unique constraints_

Better?

Reply:
Yepp. Maybe include the trivial/derivable terminology from above.